src/SolrQuery.php

```diff
@@ -33,7 +33,8 @@ public function jsonSerialize(): mixed
             $data['facet'] = [];
             $limit = $params['facet.limit'] ?? null;
             foreach ($params['facet.field'] as $field) {
-                $data['facet'][$field] = ['type' => 'terms', 'field' => $field];
+                $sort = $params["facet.sort.$field"][0] != '' ? $params["facet.sort.$field"][0] : 'count';
```

Comment:
If you decide to use `setParam` instead of `addParam`, remember to update this line.

Reply:
You should handle the case where `$params` has no key `"facet.sort.$field"`, to avoid PHP warnings

Reply:
You should delete the key `"facet.sort.$field"` from `$params` otherwise it will be sent to Solr.

Reply:
and this too